backend/src/modules/product/product.controller.ts

```diff
@@ -0,0 +1,25 @@
+import { Controller , Post, Get, Patch, Body, Query } from '@nestjs/common';
+
+import { ProductService } from './product.service';
+import { CreateProductDto } from './dto/create-product.dto';
+import { Product } from './product.interface';
+
+@Controller('api/products')
+export class ProductController {
+  constructor(private readonly productService: ProductService) {}
+
+  @Post()
+  async create(@Body() createProductDto: CreateProductDto): Promise<Product> {
+    return this.productService.create(createProductDto);
+  }
+
+  @Get()
+  async findAll(@Query() query): Promise<Product[]> {
+    return this.productService.findAll(+query.skip, +query.limit);
+  }
+
+  @Patch(':id')
+  async update(id: string, @Body() updateProductDto: CreateProductDto): Promise<Product> {
```

Comment:
Don't sure it will work as expected.
https://docs.nestjs.com/controllers.
I mean missing 'param' decorator (@Param())